src/core/memory.d

```diff
@@ -851,12 +853,15 @@ void* pureRealloc(void* ptr, size_t size) pure @nogc nothrow
     }
     return ret;
 }
+/// ditto
+void pureFree(void* ptr) @system pure @nogc nothrow
```

Comment:
Yah, let's set and restore it for now regardless. We may change that later as an optimization. Thanks!